src/sema/types.rs

```diff
@@ -399,30 +378,103 @@ fn resolve_contract<'a>(
                 delay.events.push(ResolveEventFields {
                     event_no,
                     pt,
-                    comments: tags,
+                    comments: parts.doccomments.clone(),
                     contract: Some(contract_no),
                 });
             }
             pt::ContractPart::TypeDefinition(ty) => {
-                let tags = parse_doccomments(comments, doc_comment_start, ty.loc.start());
+                annotions_not_allowed(&parts.annotations, "type", ns);
 
-                type_decl(ty, file_no, &tags, Some(contract_no), ns);
-            }
-            pt::ContractPart::FunctionDefinition(f) => {
-                if let Some(pt::Statement::Block { loc, .. }) = &f.body {
-                    doc_comment_start = loc.end();
-                    continue;
-                }
+                type_decl(ty, file_no, &parts.doccomments, Some(contract_no), ns);
             }
             _ => (),
         }
-
-        doc_comment_start = parts.loc().end();
     }
 
     broken
 }
 
+/// Resolve annotations attached to a contract
+fn contract_annotations(
+    contract_no: usize,
+    annotations: &[&pt::Annotation],
+    ns: &mut ast::Namespace,
+) {
+    let mut seen_program_id = None;
+
+    for note in annotations {
```

Comment:
What do you think of this?

```rust
if note.id.name != "progam_id" || ns.target != Target::Solana {
 ns.diagnostics.push(Diagnostic::error(
                 note.loc,
                 format!(
                     "unknown annotation '{}' on contract {}",
                     note.id.name, ns.contracts[contract_no].name,
                 ),
             ));
             continue;
}
// match statement here
```

Do you think this suggestion makes your code more organized?

Reply:
Yes, you're right. This makes it nicer. I've also put the `ns.target` comparison first.

Thanks!